File: Programmes-Qt/WSListener/anomaliesonore.cpp
#include "anomaliesonore.h"

AnomalieSonore::AnomalieSonore(double niveau, QDateTime date_debut, QDateTime date_fin, unsigned int id_as, bool debug):
    m_id_as(id_as),
    m_niveau(niveau),
    m_date_debut(date_debut),
    m_date_fin(date_fin),
    m_debug(debug){

}

AnomalieSonore::AnomalieSonore(){

}

unsigned int AnomalieSonore::get_id_as() const{
    return m_id_as;
}

double AnomalieSonore::get_niveau() const{
    return m_niveau;
}

QDateTime AnomalieSonore::get_date_debut() const{
    return m_date_debut;
}

QDateTime AnomalieSonore::get_date_fin() const{
    return m_date_fin;
}

File: monitoring_activite/objects/anomalie_sonore.php
<?php

class AnomalieSonore{

	private $_id_as;
	private $_date_debut;
	private $_date_fin;
	private $_niveau;
	private $_seuil;
	private $_duree;
	private $_plage_horaire;

	//Constructeur
	public function __construct(array $anomalie_sonore){
		$this->_id_as = (int) $anomalie_sonore['id_as'];
		$this->_date_debut = $anomalie_sonore['date_debut'];
		$this->_date_fin = $anomalie_sonore['date_fin'];
		$this->_niveau = (float) $anomalie_sonore['niveau'];
		$this->_seuil = (int) $anomalie_sonore['seuil'];
		$this->_duree = (int) $anomalie_sonore['duree'];
		$this->_plage_horaire = $anomalie_sonore['plage_horaire'];
	}

	//Getters
	public function id_as() { return $this->_id_as; }
	public function date_debut() { return $this->_date_debut; }
	public function date_fin() { return $this->_date_fin; }
	public function niveau() { return $this->_niveau; }
	public function seuil() { return $this->_seuil; }
	public function duree() { return $this->_duree; }
	public function plage_horaire() { return $this->_plage_horaire; }

}
?>
